app/controllers/dances_controller.rb

```diff
@@ -38,20 +39,36 @@ def update
   end
 
   def destroy
-    if creator?
-      @dance.destroy
-      redirect_to root_path
-    else
-      redirect_to dance_path(@dance)
+    if comments?
+      @dance.dance_comments.destroy
     end
+    @dance.destroy
+    redirect_to root_path
   end
 
   private
 
+  def creator?
+    current_user && current_user.id == @dance.user_id
+  end
+
   def set_dance
     @dance = Dance.find(params[:id])
   end
 
+  def dances
+    @dances ||= Dance.all
+  end
+
+  def comments?
+    @dance.dance_comments
+  end
```

Comment:
Hmm I would be wary of this. If I recall, if there are no associated dance_comments, activerecord will return `[]` which is still a truthy value.